website/docs/reference/programmatic-invocations.md

```diff
@@ -0,0 +1,105 @@
+
+| Scenario                                                                                    | CLI Exit Code | `success` | `result`         | `exception` |
+|---------------------------------------------------------------------------------------------|---------------|-----------|-------------------|-------------|
+| Invocation completed without error                                                          | 0             | `True`      | varies by command | `None`        |
+| Invocation completed with at least one handled error (e.g. test failure, model build error) | 1             | `False`     | varies by command | `None`        |
+| Unhandled error. Invocation did not complete, and returns no results.                       | 2             | `False`     | `None`              | Exception   |
+
+## Commitments & Caveats
+
+From dbt Core v1.5 onward, we making an ongoing commitment to providing a Python entry point at functional parity with dbt-core's CLI. We reserve the right to change the underlying implementation used to achieve that goal. We expect that the current implementation will unlock real use cases, in the short & medium term, while we work on a set of stable, long-term interfaces that will ultimately replace it.
+
+In particular, the objects returned by each command in `dbtRunnerResult.result` are not fully contracted, and therefore liable to change. Some of the returned objects are partially documented, because they overlap in part with the contents of [dbt artifacts](dbt-artifacts). As Python objects, they contain many more fields and methods than what's available in the serialized JSON artifacts. These additional fields and methods should be considered **internal and liable to change in future versions of dbt-core.**
+
+## Advanced usage patterns
+
+:::caution
+The syntax and support for these patterns are liable to change in future versions of `dbt-core`.
+:::
+
+The goal of `dbtRunner` is to offer parity with CLI workflows, within a programmatic environment. There are a few advanced usage patterns that extend what's possible with the CLI.
+
+### Reusing objects
+
+Pass pre-constructed objects into `dbtRunner`, to avoid recreating those objects by reading files from disk. Currently, the only object supported is the `Manifest` (project contents).
+
+```python
+from dbt.cli.main import dbtRunner, dbtRunnerResult
+from dbt.contracts.graph.manifest import Manifest
+
+# use 'parse' command to load a Manifest
+res: dbtRunnerResult = dbtRunner().invoke(["parse"])
+manifest: Manifest = res.result
+
+# introspect manifest
+# e.g. assert every public model has a description
+for node in manifest.nodes.values():
+    if node.resource_type == "model" and node.access == "public":
+        assert node.description != ""
+
+# reuse this manifest in subsequent commands to skip parsing
+dbt = dbtRunner(manifest=manifest)
+cli_args = ["run", "--select", "tag:my_tag"]
+res = dbt.invoke(cli_args)
+```
```

Comment:
This example is verified to work for me.

Using the following assertion made it more clear which model is missing a `description`:
```
        assert node.description != "", f"{node.name} is missing a description"
```

Reply:
@dbeatty10 That is nicer! Do you want to suggest it, or commit directly? Mi PR es tu PR